src/amoc_throttle/amoc_throttle_process.erl

```diff
@@ -201,25 +202,34 @@ async_runner(Fun) ->
 timeout(State) ->
     State#state.interval + ?DEFAULT_MSG_TIMEOUT.
 
-inc_n(#state{n = N, max_n = MaxN} = State) ->
+inc_n(#state{name = Name, n = N, max_n = MaxN} = State) ->
     NewN = N + 1,
     case MaxN < NewN of
         true ->
             PrintableState = printable_state(State),
-            ?LOG_ERROR("~nthrottle process ~p: invalid N (~p)~n", [self(), PrintableState]),
+            Msg = <<"throttle proccess has invalid N">>,
+            amoc_telemetry:execute_log(
+              error, [throttle, process], #{name => Name, n => NewN, state => PrintableState}, Msg),
             State#state{n = MaxN};
         false ->
             State#state{n = NewN}
     end.
 
-log_state(Msg, State) ->
+-spec internal_event(binary(), state()) -> any().
+internal_event(Msg, #state{name = Name} = State) ->
     PrintableState = printable_state(State),
-    ?LOG_DEBUG("~nthrottle process ~p: ~s (~p)~n", [self(), Msg, PrintableState]).
+    amoc_telemetry:execute_log(
+      debug, [throttle, process], #{self => self(), name => Name, state => PrintableState}, Msg).
+
+-spec internal_error(binary(), atom(), non_neg_integer()) -> any().
+internal_error(Msg, Name, Rate) ->
```

Comment:
I think if we also must add interval into metadata, cause rate value on it's own doesn't make much sense.